src/main/java/com/google/api/codegen/java/JavaTypeTable.java

```diff
@@ -0,0 +1,126 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.java;
+
+import com.google.api.codegen.LanguageUtil;
+import com.google.api.codegen.util.TypeAlias;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class JavaTypeTable {
+  /**
+   * A bi-map from full names to short names indicating the import map.
+   */
+  private final BiMap<String, String> imports = HashBiMap.create();
+
+  /**
+   * A map from simple type name to a boolean, indicating whether its in java.lang or not. If a
+   * simple type name is not in the map, this information is unknown.
+   */
+  private final Map<String, Boolean> implicitImports = Maps.newHashMap();
+
+  private static final String JAVA_LANG_TYPE_PREFIX = "java.lang.";
+
+  /**
+   * A map from unboxed Java primitive type name to boxed counterpart.
+   */
+  private static final ImmutableMap<String, String> BOXED_TYPE_MAP =
```

Comment:
Define constants for these, especially the keys which are also used in `ModelToJavaType.PRIMITIVE_TYPE_MAP`.